### PR TITLE
[oneDPL] Specify copy and move semantics for device_policy

### PR DESCRIPTION
**Pull request by akukanov**
Addresses #271.

For this patch, I believe the 7 days review period [applies](https://github.com/uxlfoundation/uxl_operational_procedures/blob/release/Process_Documents/Organization_Operational_Process.md#review--approval-process):

> 7 days for major changes to non-API modifications (such as changes to the documentation of an API)

### CODE REVIEW

File: source/elements/oneDPL/source/parallel_api/execution_policies.rst [hunk 1]
```diff
@@ -101,15 +102,23 @@ device_policy Class
...
 
 The ``KernelName`` template parameter, also aliased as ``kernel_name`` within the class template,
 is to explicitly provide a name for SYCL kernels executed by an algorithm the policy is passed to.
 
+The ``device_policy`` type is copy constructible, copy assignable, move constructible and move assignable.
+A policy instance constructed as or assigned a copy of another instance is associated with
+a ``sycl::queue`` that compares equal, as defined by `SYCL`_, to the queue of that other instance.
+A policy instance constructed as or assigned a move of another instance is associated with the queue
+of that other instance without copying it. It is unspecified whether the original policy is associated
+with any queue after the move. [*Note*: The predefined ``dpcpp_default`` policy object is immutable
```
Comment (andreyfe1) [inline source/elements/oneDPL/source/parallel_api/execution_policies.rst:119]: I think you can write a **Note** as it's done here: https://github.com/uxlfoundation/oneAPI-spec/blob/main/source/elements/oneDPL/source/index.rst

**Review by andreyfe1 — APPROVED** [review]
LGTM. Thanks!

File: source/elements/oneDPL/source/parallel_api/execution_policies.rst (hunk 1 above)
Comment (akukanov) [reply source/elements/oneDPL/source/parallel_api/execution_policies.rst:119]: Thanks for reviewing Andrey.

For a short clarification that depends on the context of the preceding wording an "inlined" note (similar to what the C++ standard uses) seems better to me. I will think more of it though.

Comment (andreyfe1) [reply source/elements/oneDPL/source/parallel_api/execution_policies.rst]: Thanks. It looks nicer now

File: source/elements/oneDPL/source/parallel_api/execution_policies.rst [hunk 2]
```diff
@@ -101,15 +99,22 @@ device_policy Class
...
 
 The ``KernelName`` template parameter, also aliased as ``kernel_name`` within the class template,
 is to explicitly provide a name for SYCL kernels executed by an algorithm the policy is passed to.
 
+The ``device_policy`` type is copy constructible, copy assignable, move constructible and move assignable.
+A policy instance constructed as or assigned a copy of another instance is associated with
+a ``sycl::queue`` that compares equal, as defined by `SYCL`_, to the queue of that other instance.
+A policy instance constructed as or assigned a move of another instance is associated with the queue
+of that other instance without copying it. It is unspecified whether the original policy is associated
+with any queue after the move.
```
Comment (danhoeflinger) [inline source/elements/oneDPL/source/parallel_api/execution_policies.rst:116]:
```suggestion
of that other instance without copying it. It is unspecified whether the original policy is associated
with any queue after being moved from.
```
I don't really like use "the move" in this context without first introducing the move we are referring to more directly.

**Review by danhoeflinger — APPROVED** [review]
LGTM, small wording suggestion, not required for me.

File: source/elements/oneDPL/source/parallel_api/execution_policies.rst (hunk 2 above)
Comment (akukanov) [reply source/elements/oneDPL/source/parallel_api/execution_policies.rst:116]: Would this be better?

> It is unspecified whether the moved-from policy instance is associated with any queue after the move.

Comment (danhoeflinger) [reply source/elements/oneDPL/source/parallel_api/execution_policies.rst:116]: yes that is better

**Review by timmiesmith — APPROVED** [review]
This looks good to me.

**Pull request merged**